topsy.el

```diff
@@ -104,6 +114,22 @@ Return non-nil if the minor mode is enabled."
 
 ;;;; Functions
 
+(defun topsy--header-string ()
+  "Return string found by `topsy-fn' or line above window start."
+  (or (when-let ((header (and topsy-fn (funcall topsy-fn))))
```

Comment:
This could be written in terms of `if-let` rather than `(or (when-let ...`.  What do you think?